packages/thirdweb/src/react/web/ui/ConnectWallet/Modal/ConnectModalContent.tsx

```diff
@@ -94,7 +96,26 @@ export const ConnectModalContent = (props: {
       onGetStarted={() => {
         setScreen(reservedScreens.getStarted);
       }}
-      selectWallet={setScreen}
+      selectWallet={(newWallet) => {
+        setScreen(newWallet);
+
+        // When selecting Coinbase, fire off the connection immediately
+        // this fixes an issue on safari where the CB popup window would get blocked
+        // because of the delay between this click and the actual connection
+        // that happens within 'CoinbaseSDKConnection' component 2 layers down
+        // when the cb extension is installed -> pops the extension
+        // when the extension is not installed -> pops a new window
+        // then in 'CoinbaseSDKConnection' the connection is requested again
+        // which just "links" to the extension / popup window a few ms later
+        // and handle the completion / error handling there
+        // this connection flow structure needs to be refactored to properly handle this
+        if (newWallet.id === COINBASE) {
```

Comment:
does this properly set up all the listeners etc?

Reply:
this comes with more problems, here's what I've found so far:

1. coinbase wallet will now call connect twice (here, and in the useEffect below)
2. this causes that "second" connection to *always* fail -> we always end up in an error screen

to resolve this we really have 2 options:
1. refactor this behavior so *all* wallets connect on-click (if they have distinct connect behaviors that can be known at click-time) -> then pass the wallet instance that is connecting down to the screen, which can subscribe to the `onConnect` listener (we also likely need to expose an `onConnectionError` listener to handle that case) -> that screen becomes purely the "retry" and "visualize the state" screen instead of ever triggering a connection
2. add an explicit button to that screen to trigger the connection, means every user has to make an extra click but is likely an easier refactor - again we'd want to do this for every wallet not just coinbase

Reply:
solved upstack